architecture/ambient/ztunnel.md

```diff
@@ -0,0 +1,227 @@
+Istio's AuthorizationPolicy has label selectors.
+However, we intentionally do not send those as part of the Workload API, in order to keep the size low.
+
+The obvious solution to this is to put the list of selected workloads into the policy itself.
+However, this means anytime a workload changes (often), we need to update the policy.
+
+Instead, the opposite was chosen: each workload will list the policies that select it.
+This works out to be more efficient in common cases where policies change must less often than workloads.
+This only applies for selector-based policies; namespaced and global policies can be handled without needing to list them out.
+
+## Redirection
+
+As ztunnel aims to transparently encrypt and route users traffic, we need a mechanism to capture all traffic entering and leaving "mesh" pods.
+This is a security critical task: if the ztunnel can be bypassed, authorization policies can be bypassed.
+
+Redirection must meet these requirements:
+* All traffic *egressing* a pod in the mesh should be redirected to the node-local ztunnel on port 15001.
+  * It is critical that this path preserves the Service IP, if the traffic was to a Service.
+* All traffic *ingressing* a pod on port 15008 in the mesh should be redirected to the node-local ztunnel on port 15008 (more on this later).
+* All other traffic *ingressing* a pod in the mesh should be redirected to the node-local ztunnel on port 15006.
+
+TODO: fill in implementation details of how redirection is actually implemented.
+
+## HBONE
+
+Along with pass-through traffic, Ztunnel supports the "HBONE" protocol.
+This is not really so much a new protocol, but rather a name we came up with to refer to the expectations of clients and servers communicating in the mesh.
+
+HBONE is just a standard HTTP `CONNECT` tunnel, over mutual TLS with mesh (SPIFFE) certificates, on a well known port (15008).
+The target destination address is set in the `:authority` header, and additional headers can be included as well.
+Currently, only HTTP/2 is supported, though HTTP/1.1 and HTTP/3 are planned.
+
+Currently, SNI is not set by Istio clients and ignored by Istio servers.
+This makes identifying which certificate to use problematic for Ztunnel.
+To handle this, requests to Ztunnel are sent to `DestinationPod:15008` and redirected to ztunnel, rather than `ZtunnelPod:15008`.
+The original destination is then extracted to determined which certificate to use.
+SNI is not used because it is illegal to use IP's in SNI, and there is no other existing standard format to represent what we need to.
+Additionally, using the redirection mechanism reduces the need for clients to know the destination's ztunnel address.
+
+### Pooling
+
+User connections can be multiplexed over shared HBONE connections.
+This is done through standard HTTP/2 pooling.
+The pooling is keyed off the `{source identity, destination identity, destination ip}`.
```

Comment:
Unrelated to the PR, but does this mean that unauthed requests to a specific destination workload in the mesh are all handled via one connection? Is there a mechanism to relieve pressure if someone attempts to DDoS?

Reply:
Unauthed would be `external client ---plaintext--> destination ztunnel ---plaintext --> destination workload`. So there can be no multiplexing.

Multiplexing only occurs with HBONE.

Now if we do the "simple TLS HBONE hairpin" then presumably we would pool that under one connection which we may want to limit

Reply:
Ah right. Thanks for the clarification